FoodPin/FoodPin/Controller/RestaurantDetailViewController.swift

```diff
@@ -9,21 +9,25 @@ import UIKit
 
 class RestaurantDetailViewController: UIViewController {
 
-    @IBOutlet private var restaurantImageView: UIImageView!
-    @IBOutlet private var restaurantLocation: UILabel!
-    @IBOutlet private var restaurantType: UILabel!
-    @IBOutlet private var restaurantName: UILabel!
+    @IBOutlet var tableView: UITableView!
+    @IBOutlet var headerView: RestaurantDetailHeaderView!
     private var restaurant: Restaurant = Restaurant.init()
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        navigationItem.largeTitleDisplayMode = .never
+        navigationController?.navigationBar.prefersLargeTitles = false
 
-        restaurantImageView.image = UIImage(named: restaurant.getImage())
-        restaurantName.text = restaurant.getName()
-        restaurantType.text = restaurant.getType()
-        restaurantLocation.text = restaurant.getLocation()
+        // Configure header view
+        headerView.configure(nameLabel: restaurant.getName(), typeLabel: restaurant.getType(),
+                             headerImageView: restaurant.getImage())
+        let heartImage = restaurant.getIsFavorite() ? "heart.fill" : "heart"
+        headerView.setHeartButton(color: restaurant.getIsFavorite() ? .systemYellow : .white)
+        headerView.setHeartButton(image: heartImage)
+
```

Comment:
Maybe you can pass the restaurant as a variable to the configure method and get everything you need from there in  RestaurantDetailHeaderView and set the heart button there too